docs/guides/05-building/cross-platform.md

```diff
@@ -4,4 +4,144 @@ sidebar_position: 2
 
 # Cross-Platform Compilation
 
-How to use GH Action for Building: a glance at Tauri Action.
+Tauri relies heavily on native libraries and toolchains, so meaningful cross-compilation is not possible at the current moment. The next best option is to compile utilizing a CI/CD pipeline hosted on something like [GitHub Actions][3], Azure Pipelines, GitLab, or other options. The pipeline can run the compilation for each platform simultaneously making the compilation and release process much easier.
+
+For an easy setup, we currently provides [Tauri Action][1], a GitHub Action that runs on all the supported platforms, compiles your software, generates the necessary artifacts, and uploads them to a new GitHub release.
+
+## Tauri GitHub Action
+
+
+Tauri Action leverages GitHub Actions to simultaneously build your application as a Tauri native binary for macOS, Linux, and Windows, and automates creating a GitHub release.
+
+This GitHub Action may also be used as a testing pipeline for your Tauri app, guaranteeing compilation runs fine on all platforms for each pull request sent, even if you don't wish to create a new release.
+
+### Getting Started
+
+To set up Tauri Action you must first set up a GitHub repository. You can use this action on a repo that doesn't have Tauri configured since it automatically initialize Tauri before building and configure it to use your artifacts.
+
+Go to the Actions tab on your GitHub project and choose "New workflow", then choose "Set up a workflow yourself". Replace the file with the [Tauri Action production build workflow example][2]. Alternatively, you may set up the workflow based on the [example lower on this page][6]
+
+### Configuration
+
+You can configure Tauri with the `configPath`, `distPath` and `iconPath` options.
+
+Custom Tauri CLI scripts can be run with the `tauriScript` option. So instead of running `yarn tauri build` or `npx tauri build`, `${tauriScript}` will be executed. This can be useful when you need custom build functionality such as when creating Tauri apps e.g. a `desktop:build` script.
+
+When your app isn't on the root of the repo, use the `projectPath` input.
+You may modify the workflow name, change the triggers, and add more steps such as `npm run lint` or `npm run test`. The important part is that you keep the below line at the end of the workflow, since this runs the build script and releases the artifacts:
+
+```yaml
+- uses: tauri-apps/tauri-action@v0
+```
+
+### GitHub Environment Token
```

Comment:
Can we move this section down below the big code example? It makes more sense in the context there IMO, it feels a bit weird that this additional info about the token would be more important that the *How to trigger* section or code example